src/component/bom.js

```diff
@@ -1873,6 +1876,50 @@ const osInfoReason = (resource) => {
 }
 
 
+const osInfoReasonE = (resource) => {
+  const eolString = (eolInfo) => {
+    if (typeof eolInfo == 'string') return `EOL reached on ${new Date(eolInfo).toLocaleDateString()}`
```

Comment:
IIRC, eolInfo will be null (or was it undefined?) if no data is available. If this is correct, then I suggest to not check for type being string, but to instead check for the value being null/undefined (or: falsy)

Also, from the code, it appears this function is expected to be an expiryDate. hence, I suggest to rename the function's parameter accordingly